engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/manager/ImageStoreProviderManagerImpl.java

```diff
@@ -146,7 +146,13 @@ public ImageStoreEntity getImageStore(String uuid) {
     @Override
     public DataStore getImageStore(List<DataStore> imageStores) {
         if (imageStores.size() > 1) {
-            Collections.shuffle(imageStores); // Randomize image store list.
+            Collections.sort(imageStores, new Comparator<DataStore>() {
+                @Override
```

Comment:
This still does not cleanly solve the design issue. The `getImageStore` may be used in several places and simply randomized return (whether shuffled or sorted by capacity) does not make much difference. Instead, a check may be performed preferably via a different method that checks if image store has enough capacity, that check method may be used either by this method or the consumers of `getImageStore`.

Reply:
The idea behind Sorting by capacity was that the imagestore with the most available capacity will always be on the top of the list hence selected first.
I thought about checking if imagestore has enough capacity but that would mean that I compare the image size vs remaining capacity of the image store, that of course would mean that I introduce another parameter to this `getImageStore` method, perhaps Image id or something, to check for size of image. Not sure how this will go down with the consumers of `getImageStore`.